src/legacy/map-to-non-cylic-graph.ts

```diff
@@ -0,0 +1,80 @@
+import { DepGraph, DepGraphBuilder, PkgInfo } from '..';
+import { DepGraphInternal, NodeInfo } from '../core/types';
+import { EventLoopSpinner } from './event-loop-spinner';
+
+type NodeId = string;
+
+export async function mapToNonCyclicGraph(
+  depGraphToMap: DepGraph,
+): Promise<DepGraph> {
+  const depGraph = depGraphToMap as DepGraphInternal;
+  if (!depGraph.hasCycles()) {
+    return depGraph;
+  }
+
+  const depGraphBuilder = new DepGraphBuilder(
+    depGraph.pkgManager,
+    depGraph.rootPkg,
+    depGraph.rootNodeId
+  );
+
+  const eventLoopSpinner = new EventLoopSpinner();
```

Comment:
This can move to module scope - there's no value in passing a locally scoped instance around